src/manifolds/HyperbolicHyperboloid.jl

```diff
@@ -242,8 +246,8 @@ function exp!(M::Hyperbolic, q, p, X, t::Number)
 end
 function exp!(M::Hyperbolic, q, p, X)
     vn = sqrt(max(inner(M, p, X, X), 0.0))
-    vn < eps(eltype(p)) && return copyto!(q, p)
-    q .= cosh(vn) .* p .+ (sinh(vn) / vn) .* X
+    sn = vn == 0.0 ? 1.0 : sinh(vn) / vn
```

Comment:
```suggestion
    sn = vn == 0 ? one(vn) : sinh(vn) / vn
```
In case `vn` is of a different type.